compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -533,3 +539,140 @@ private val PropertyDescriptor.declaresDefaultValue: Boolean
         is KtParameter -> declaration.defaultValue != null
         else -> false
     }
+
+fun <T : Annotation> KSAnnotated.getAnnotation(annotationKClass: KClass<T>): T? {
+    return this.annotations.firstOrNull {
+        annotationKClass.qualifiedName == it.annotationType.resolve().declaration.qualifiedName?.asString()
+    }?.toAnnotation(annotationKClass)
+}
+
+fun KSAnnotated.getAnnotations(): Sequence<out Annotation> {
+    return this.annotations.map {
+        val kClass = Class.forName(it.annotationType.resolve().declaration.qualifiedName?.asString()).kotlin
+        it.toAnnotation(kClass as KClass<Annotation>)
```

Comment:
What if some of the annotations aren't available at compile time? I.e., when `Class.forName` fails.

IMHO, a closer analogy to `KSAnnotated` in Java API might be [AnnotatedConstruct](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/AnnotatedConstruct.html), rather than `AnnotatedElement`. If we really want all annotations without specifying `KClass` or `Class` (being able to write that in processor ensures their availability at processor's runtime / user code's compile time), we need to define the behavior. I recommend removing this in this PR and add it in a later PR.

Reply:
Maybe I misunderstand, but wouldn't the Annotation classes have to be available if we were compiling the codebase, otherwise wouldn't it fail compilation in general?

Reply:
Or is it possible to have a different compile classpath vs the processors classpath? If so then that makes sense to remove this.

Reply:
Yes. For example, the processor won't be able to instantiate `A`, because it isn't even compiled yet. I think that is why methods in `AnnotatedConstruct` require a `Class<A>`; that makes sure `A` is instantiable in processor.

```
// Same file
annotation class A

@A
class C
```

Reply:
done